### PR TITLE
http-client-java, union variant in union

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
found in https://github.com/joseharriaga/openai-in-typespec

### CODE REVIEW

File: packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/util/ClientModelUtil.java [hunk 1]
```diff
@@ -645,7 +645,11 @@ public static boolean includePropertyInConstructor(ClientModelProperty property,
         // the constructor.
         boolean polymorphicDiscriminatorIsRequired = property.isPolymorphicDiscriminator() && property.isRequired();
 
-        return requiredAndIncluded && (notReadOnlyOrIncludeReadOnly || polymorphicDiscriminatorIsRequired);
+        boolean notConstant = !property.isConstant();
+
+        return requiredAndIncluded
+            && (notReadOnlyOrIncludeReadOnly || polymorphicDiscriminatorIsRequired)
+            && notConstant;
```
Comment (weidongxu-microsoft) [inline packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/util/ClientModelUtil.java:652]: I think this be a bug. Constant property should not be in ctor.

**Comment by haolingdong-msft** [comment]
I don't quite get the case here, could you paste or point out the case?

**Comment by weidongxu-microsoft** [comment]
> I don't quite get the case here, could you paste or point out the case?

Without this, generated code be

```java
    @Generated
    private String object = "realtime.item";

    @Generated
    protected RealtimeResponseItem(String id) {
        this.id = id;
    }

    @Generated
    private RealtimeResponseFunctionCallItem(String object, String id, String name, String callId, String arguments,
        RealtimeItemStatus status) {
        super(object, id);
    ...
```

It won't compile.

**Comment by weidongxu-microsoft** [comment]
I cannot find the "UnionVariant" case. Too many union (many of them is anonymous) in openai repo.

**Review by haolingdong-msft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**